docs/block-development/tutorials/integrating-protection-checkout-block.md

```diff
@@ -0,0 +1,173 @@
+        },
+      });
+
+      unsubscribe();
+    }
+  }, "wc/store/cart" );
+}
+```
+
+**Key points about this JavaScript:**
+
+- We subscribe to the cart data store to detect when the checkout is ready
+- The `turnstile.render()` method initializes your CAPTCHA (replace with your specific implementation)
+- `setExtensionData()` stores the token in the checkout data store
+- The namespace should be unique to your plugin (e.g., `my-plugin-turnstile`)
+
+### Data Store Integration
+
+For more information about the checkout data store and available methods, see the [Checkout Data Store documentation](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/docs/third-party-developers/extensibility/data-store/checkout.md).
+
+## Step 3: Server-Side Validation
+
+The most critical step is validating the protection token on the server side. This should happen early in the authentication process to prevent any unauthorized checkout attempts.
+
+### Using the core `rest_authentication_errors` filter
+
+The [`rest_authentication_errors`](https://developer.wordpress.org/reference/hooks/rest_authentication_errors/) filter is the ideal place to validate your protection token because it runs before any checkout processing begins.
+
+```php
+add_filter( 'rest_authentication_errors', 'plugin_check_turnstile_token' );
+
+function plugin_check_turnstile_token( $result ) {
+    // Skip if this is not a POST request.
+    if ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] !== 'POST' ) {
+        // Always return the result or an error, never a boolean. This ensures other checks aren't thrown away like rate limiting or authentication.
+        return $result;
+    }
+
+    // Skip if this is not the checkout endpoint.
+    if ( ! preg_match( '#/wc/store(?:/v\d+)?/checkout#', $GLOBALS['wp']->query_vars['rest_route'] ) ) {
+        return $result;
+    }
+
+    // get request body
+    $request_body = json_decode( \WP_REST_Server::get_raw_data(), true );
+
+    if ( isset( $request_body['payment_method'] ) ) {
+        $chosen_payment_method = sanitize_text_field(  $request_body['payment_method'] );
+
+        // Provide ability to short circuit the check to allow express payments or hosted checkouts to bypass the check.
+        $selected_payment_methods = apply_filters(  'plugin_payment_methods_to_skip', array('woocommerce_payments' ) );
+        if( is_array( $selected_payment_methods ) ) {
+            if ( in_array( $chosen_payment_method, $selected_payment_methods, true ) ) {
+                return $result;
+            }
+        }
+    }
+
+    $extensions = $request_body['extensions'];
+    if ( empty( $extensions ) || ! isset( $extensions['plugin-namespace-turnstile'] ) ) {
+        return new WP_Error( 'challenge_failed', 'Captcha challenge failed' );
+    }
+    $token = sanitize_text_field( $extensions['plugin-namespace-turnstile']['token'] );
+    $check = my_token_check_function( $token );
```

Comment:
Should probably be explicit about this function needing to be implemented by the plugin.